.github/workflows/main.yml

```
@@ -469,6 +469,19 @@ jobs:
             echo NULL > $dummy_screenshot
           }
 
+          echo "Going to create dummy trailers & their images for Application Payload creation:"
+          $xmlDoc.ProductDescription.Trailers.Trailer | ForEach-Object {
+            $dummy_trailer = "MSAppPayload/PDPs/en-us/" + $_.FileName
+            echo $dummy_trailer
+            echo NULL > $dummy_trailer
```

What stops these dummy files from being included in the msix bundle?

Also, why create a file containing (presumably) the string `NULL\n` rather than touching empty files?

---

> What stops these dummy files from being included in the msix bundle?

The MSIX bundle does not need the dummy files.  The dummy files are prepared for the Application Payload of a submission to MS store.  The [submission](https://github.com/endlessm/endless-key-app/blob/v6.32.2/.github/workflows/main.yml#L472-L480) includes:
* MSIX/APPX bundle
* Application Payload: PDP.xml and assets
* ...

The PDP.xml lists the assets' filenames.  However, Endless Key's Alpha Test Flight does not need the assets, actually.  MS store gets those information from production directly.  But, the creation of Alpha Test Flight submission's Application Payload needs the files.  So, produce dummy files for it.

> Also, why create a file containing (presumably) the string `NULL\n` rather than touching empty files?

Does Windows, or Powershell has touch command?

---

I don't know but I [searched the web](https://duckduckgo.com/?q=powershell+create+empty+file&ia=web) and the answer is:

```suggestion
            New-Item $dummy_trailer -type file
```

But if you want to use `echo` (which in Powershell is an alias to `Write-Output`), why use the string `NULL`?

---

Simply, NULL string in the files.  And, I remember that the files should has some content for `New-SubmissionPackage` command.  Otherwise, it shows empty files error.  Tested one year ago.  I do not know it is still true for current StoreBroker, or not.